biz.aQute.bnd/test/aQute/bnd/main/DistroCommandTest.java

```diff
@@ -0,0 +1,92 @@
+package aQute.bnd.main;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.launch.Framework;
+
+import aQute.bnd.main.RemoteCommand.DistroOptions;
+import aQute.bnd.main.RemoteCommand.RemoteOptions;
+import aQute.bnd.osgi.Jar;
+import aQute.lib.getopt.CommandLine;
+import aQute.lib.io.IO;
+import junit.framework.TestCase;
+
+@SuppressWarnings("restriction")
+public class DistroCommandTest extends TestCase {
+
+	private Map<String,Object>	configuration;
+	private Framework			framework;
+	private File				tmp;
+
+	@Override
+	protected void setUp() throws Exception {
+		tmp = IO.getFile("generated/tmp");
+		tmp.mkdirs();
+
+		configuration = new HashMap<String,Object>();
+		configuration.put(Constants.FRAMEWORK_STORAGE_CLEAN, Constants.FRAMEWORK_STORAGE_CLEAN_ONFIRSTINIT);
+		configuration.put(Constants.FRAMEWORK_STORAGE, new File(tmp, "fwstorage").getAbsolutePath());
+		configuration.put(Constants.FRAMEWORK_SYSTEMPACKAGES_EXTRA, "org.osgi.framework.launch;version=1.4");
+		framework = new org.apache.felix.framework.FrameworkFactory().newFramework(configuration);
```

Comment:
So on second thought I will need some help with this.  Can you point me to an example of where the Launcher is used to start the framework instead of just calling it directly?  thanks.